src/core/Form/Checkbox/Checkbox.tsx

```diff
@@ -66,50 +118,136 @@ class DefaultCheckbox extends Component<CheckboxProps> {
 
   render() {
     const {
-      children,
+      id,
+      className,
       disabled = false,
-      onClick,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledBy,
+      'aria-describedby': ariaDescribedBy,
+      children,
+      checked: dismissChecked,
+      defaultChecked: dismissDefaultChecked,
+      onClick: dismissOnClick,
+      hintText,
       status,
+      statusText,
+      name,
+      value,
+      onClick,
       variant,
-      className,
       ...passProps
-    } = withSuomifiDefaultProps(this.props);
+    } = this.props;
+
     const { checkedState } = this.state;
+
+    if (!children) {
+      logger.error(
+        'Checkbox component should have a label or a child element that acts as one. Add label content or a child element.',
+      );
+    }
+
+    if (
+      ('name' in this.props && (typeof name !== 'string' || name === '')) ||
+      ('value' in this.props && (typeof value !== 'string' || value === ''))
+    ) {
+      logger.warn(
+        'Name and value props should have non-empty values if provided.',
+      );
+    }
+
+    const statusTextId = `${id}-statusText`;
+    const hintTextId = `${id}-hintText`;
+
+    const getDescribedBy = () => {
+      if (statusText || hintText || ariaDescribedBy) {
+        return {
+          'aria-describedby': [
+            ...(statusText ? [statusTextId] : []),
+            ...(hintText ? [hintTextId] : []),
+            ...(ariaDescribedBy ? [ariaDescribedBy] : []),
+          ].join(' '),
+        };
+      }
+      return {};
+    };
```

Comment:
This is done like this to prevent empty spaces in classname if some are not given.